lib/pbench/test/unit/server/test_indexing_tarballs.py

```diff
@@ -0,0 +1,317 @@
+
+    def make_all_actions(self) -> JSONARRAY:
+        __class__.make_all_called += 1
+        return [{"action": "make_all_actions", "name": self.name}]
+
+    @classmethod
+    def reset(cls):
+        cls.make_tool_called = 0
+        cls.make_all_called = 0
+
+
+class FakeSync:
+    tarballs: Dict[Operation, List[Dataset]] = {}
+    called: List[str] = []
+    did: Optional[Operation] = None
+    updated: Optional[List[Operation]] = None
+    errors: Optional[str] = None
+
+    @classmethod
+    def reset(cls):
+        cls.tarballs = {}
+        cls.called = []
+        cls.did = None
+        cls.updated = None
+        cls.errors = None
+
+    def __init__(self, logger: Logger, component: str):
+        self.logger = logger
+        self.component = component
+
+    def next(self, operation: Operation) -> List[Dataset]:
+        __class__.called.append(f"next-{operation.name}")
+        assert operation in __class__.tarballs
+        return __class__.tarballs[operation]
+
+    def update(
+        self,
+        dataset: Dataset,
+        did: Optional[Operation],
+        enabled: Optional[List[Operation]],
+    ):
+        __class__.did = did
+        __class__.updated = enabled
+
+    def error(self, dataset: Dataset, message: str):
+        __class__.errors = message
+
+
+class FakeController:
+    def __init__(self, path: Path, incoming: Path, results: Path, logger: Logger):
+        self.name = path.name
+        self.path = path
+        self.incoming = incoming / self.name
+        self.results = results / self.name
+        self.logger = logger
+
+
+class FakeTarball:
+    def __init__(self, path: Path, controller: FakeController):
+        self.name = path.name
+        self.tarball_path = path
+        self.controller = controller
+        self.unpacked = f"/incoming/{path.name}"
+
+
+class FakeCacheManager:
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        self.config = config
+        self.logger = logger
+        self.datasets = {}
+
+    def find_dataset(self, resource_id: str):
+        controller = FakeController(
+            Path("/archive/ctrl"), Path("/incoming"), Path("/results"), self.logger
+        )
+        return FakeTarball(
+            Path(f"/archive/ctrl/tarball-{resource_id}.tar.xz"), controller
+        )
+
+
+@pytest.fixture()
+def mocks(monkeypatch, make_logger):
+    FakeDataset.logger = make_logger
+    with monkeypatch.context() as m:
+        m.setattr("pbench.server.indexing_tarballs.Sync", FakeSync)
+        m.setattr("pbench.server.indexing_tarballs.PbenchTarBall", FakePbenchTarBall)
+        m.setattr("pbench.server.indexing_tarballs.Report", FakeReport)
+        m.setattr("pbench.server.indexing_tarballs.Dataset", FakeDataset)
+        m.setattr("pbench.server.indexing_tarballs.Metadata", FakeMetadata)
+        m.setattr("pbench.server.indexing_tarballs.CacheManager", FakeCacheManager)
+        yield m
+    FakeSync.reset()
+    FakePbenchTarBall.reset()
```

Comment:
Should we be resetting `FakeDataset` and/or `FakePbenchTemplates` here, as well?